lib/api/formtools/grid.js

```diff
@@ -0,0 +1,263 @@
+var linz = require('../../../');
+    async = require('async'),
+    clone = require('clone');
+
+(function  () {
+
+    function getSettings(modelName, cb) {
+
+        if (!modelName || typeof(modelName) !== 'string' || !modelName.length) {
+            throw new Error('modelName is required and must be of type string and cannot be emptied.');
+        }
+
+        var model = linz.get('models')[modelName];
+
+        model.getGrid(function (err, settings) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            return cb(err, clone(settings));
+
+        });
+
+    }
+
+    function getToolbarItems (req, res, modelName, cb) {
+
+        if (!modelName || typeof(modelName) !== 'string' || !modelName.length) {
+            throw new Error('modelName is required and must be of type string and cannot be emptied.');
+        }
+
+        var model = linz.get('models')[modelName];
+
+        async.waterfall([
+
+            function (callback) {
+                return getSettings(modelName, callback);
+            }
+
+        ], function (err, settings) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            if (!settings.toolbarItems.length) {
+                return cb(null);
+            }
+
+            async.each(settings.toolbarItems, function (item, itemDone) {
+
+                if (!item.renderer) {
+                    return itemDone(null);
+                }
+
+                // this is a custom action, let's execute the function to get the HTML markup
+                item.renderer(req, res, function (itemErr, html) {
+
+                    if (itemErr) {
+                        return itemDone(itemErr);
+                    }
+
+                    item.html = html;
+
+                    return itemDone(null);
+                });
+
+            }, function (itemErr) {
+                return cb(itemErr, settings.toolbarItems);
+            });
+
+        });
+
+    }
+
+    function getFilters (modelName, cb) {
```

Comment:
I've renamed:
- ```getFilters``` to ```renderFilters```
- ```getToolbarItems``` to ```renderToolbarItems```
- ```getSearchFilters``` to ```renderSearchFilters```

so the function names are more inline with the function itself.